### ISSUE TITLE
USB - Scanner for Horizon View Client

### ISSUE THREAD

**Issue by rohrbachger** [Thinstation/thinstation#625]
Is there a package available like SANE for scanner?
Probably a driver package like Twain might be enough.

The Horizon View Client is doing the redirection into the View Desktop.
So no scanner software is needed, just to the driver that 

On Ubuntu 20.04 this is working fine, so maybe it is no big deal to get it on thinstation running.

Has someone experience with scanners on thinstation?

**Comment by Thinstation** [comment]
the sane port is available. could try repackage.

**Comment by Thinstation** [comment]
Course, it could be that you don't have the modules for the scanner and you need them, or you have them and don't want them.

**Comment by rohrbachger** [comment]
For testing I have an USB scanner LIDE 110. According some documentation it should work with sane-usb.
I have tried compiling sane from source on my ubuntu 20.04, to get some understanding how it works.
Not really success yet, but as allways this takes time.
On thinstation I modified the package to get the scanner redirection installed.
For testing I run /lib/vmware/view/scannerclient/ftscanhvd.
This starts a daeomon in the background. After startinc a view session I'm getting the scanner redirectionicon in the view session.
But no scanner is there. So I guess the sane driver is needed.

**Comment by Doncuppjr** [comment]
So you need the modules for your scanner?

**Comment by rohrbachger** [comment]
That's from the documentation Vmware View
Scanner redirection supports scanning devices that are compatible with the SANE interface standard. You must install the SANE scanner device drivers on the local client system. You do not need to install the scanner device drivers on a remote desktop. 
We are using View with PCOIP, so no RDP.
That's some info I found for ubuntu: https://help.ubuntu.com/community/sane

On my standard ubuntu 20.04 it works with the view horizon client out of the box without installing something extra.

sane-find-scanner seems in standard already installed.
sane-utils is listed under aptitude, so I guess I need this and not the complete sane package.
This modul I would need.

**Comment by Thinstation** [comment]
Do an allmodules build.

**Comment by rohrbachger** [comment]
allmodules build did not help. I did not get sane drivers compiled yet.
There are two additional services in thinstation for scanner redirection necessary
ftscanhv and ftsprhv. both should start automatically, but only one is doing. It can be started manually.
Vmwares documentation does not goes to that details, Google answers are rare.
This takes time to figure out...

**Comment by Thinstation** [comment]
mkrepackage sane, then put package sane in build.conf

**Comment by rohrbachger** [comment]
This hint was perfect, and very easy to do.
Yes, I got my old Canon Lide scanner now working with Horizion View Client Version 2012 ( actual version) with thinstation.

Still I need to figure out some details but I scanned the first page!!
The ftscanhv service is not starting automatically, but this Iḿ sure I will figure out with some testing.
Also sane-find-scanner needs to run once.
Anyway I need to figure out for other scanners, how this will work.We do have brother in use.
Bu it shows that in general it works fine with thinstation.

I will make a pull request, when I have finished my investigations.
I made some small changes for the view install script.
